lib/internal/errors.js

```diff
@@ -196,21 +187,24 @@ function lazyBuffer() {
   return buffer;
 }
 
-function isErrorStackTraceLimitWritable() {
-  // Do no touch Error.stackTraceLimit as V8 would attempt to install
-  // it again during deserialization.
-  if (require('internal/v8/startup_snapshot').namespace.isBuildingSnapshot()) {
-    return false;
-  }
-
-  const desc = ObjectGetOwnPropertyDescriptor(Error, 'stackTraceLimit');
-  if (desc === undefined) {
-    return ObjectIsExtensible(Error);
+let stackTraceLimitWritable = 0;
+function setStackTraceLimit(limit) {
+  if (stackTraceLimitWritable === 1) {
+    try {
+      Error.stackTraceLimit = limit;
+    } catch {
+      stackTraceLimitWritable = -1;
```

Comment:
Nit: A comment here on why this would throw would be helpful for future generations ;-)

Reply:
I suggest to address that in a follow-up PR? :)

```
      // Setting the property in strict mode would cause this to throw an error
      // while using the --frozen-intrinsics flag due to the writable attribute
      // being set to false.
```

Reply:
I actually wonder, if we want to keep that property writable even when using frozen intrinsics.
@aduh95 @bmeck you might have an opinion about that?

Reply:
To fill the contract of `--forzen-intrinsics` (opt-out of all possible prototype mutation), we need the property to be non-configurable, it's true that it could still be writable without introducing too much risk I believe.

But even if we did that, we would probably still want the try catch in case some user code has changed this to non-writable for some reason, unless it's introducing too much maintenance burden. wdyt?